include/swift/Serialization/Validation.h

```diff
@@ -144,9 +144,12 @@ class ExtendedValidationInfo {
 /// \param[out] extendedInfo If present, will be populated with additional
 /// compilation options serialized into the AST at build time that may be
 /// necessary to load it properly.
+/// \param[out] dependencies If present, will be populated with list of
+/// input files the module depends on, if present in INPUT_BLOCK.
 ValidationInfo
 validateSerializedAST(StringRef data,
-                      ExtendedValidationInfo *extendedInfo = nullptr);
+                      ExtendedValidationInfo *extendedInfo = nullptr,
+                      SmallVectorImpl<StringRef> *dependencies = nullptr);
```

Comment:
StringRef isn't going to cut it long-term, since "newer mtime" turns out to not be a great way to tell if something's changed, but it's fine for standing it up.

Reply:
Sure. Is `pair<StringRef, uint64_t>` (name and size) adequate or do you want a full stat-like struct here?

Reply:
It looks like the driver is using name-and-mtime but just checking whether the mtime *changed.* This mechanism doesn't have to be more fine-grained than the driver, but I remember horror stories of fast machines on FAT32 partitions, so maybe they should *both* be using size *and* mtime.

Reply:
Yeah I mean I will definitely be using mtime too. But the dep's mtime-at-build-time doesn't have to be stored in the deps file since the relevant condition is "any current dep mtime newer than output mtime".

Reply:
I'm just trying to figure out whether that plus size is enough.

Reply:
"newer" is suspect, though. If I do some kind of filesystem-level rollback, the cached swiftmodule still needs to be treated as invalid. (Alternately, if the input gets read, then I change it, then the swiftmodule is written, it's *already* invalid.)

Reply:
That can definitely be a follow-up PR, though!